### PR TITLE
ci: add github pipeline

### PR DESCRIPTION
**Pull request by elysee15**

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -0,0 +1,39 @@
+name: Workflow for whisper ui
+on: [pull_request]
```
Comment (bbengfort) [inline .github/workflows/build.yml]: Could we run the tests on the following spec, please?

```
on:
  push:
    branches:
      - main
    tags:
      - 'v*'
  pull_request:
```

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -0,0 +1,39 @@
+name: Workflow for whisper ui
```
Comment (bbengfort) [inline .github/workflows/build.yml]: This name shows up in the Badge - which I'd also like to add to the README; so could we change it to:

```
name: Whisper UI Tests
```

We can add the badge to the README with the following Markdown:

```
![GitHub Actions CI](https://github.com/rotationalio/whisper-ui/actions/workflows/build.yaml/badge.svg?branch=main)
```

File: .github/workflows/build.yml [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+    name: Build
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        node-version: [12.x, 14.x]
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Install dependencies
+        run: yarn
```
Comment (bbengfort) [inline .github/workflows/build.yml]: Is yarn automatically installed on `ubuntu-latest` without needing setup?

File: .github/workflows/build.yml [hunk 4]
```diff
@@ -0,0 +1,39 @@
+name: Workflow for whisper ui
+on: [pull_request]
+
+jobs:
+  build:
+    name: Build
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        node-version: [12.x, 14.x]
```
Comment (bbengfort) [inline .github/workflows/build.yml]: Do we want to add `16.x` here?

**Review by bbengfort — APPROVED** [review]
Awesome, I'm so glad to have front-end testing and CI going - this to me is the biggest change I've made in my career; with all my other jobs we would simply hire QA folks, there were never any tests! 

Thank you so much for working through how to get GitHub Actions working - nice one!

File: .github/workflows/build.yml (hunk 1 above)
Comment (elysee15) [reply .github/workflows/build.yml]: Okay

File: .github/workflows/build.yml (hunk 2 above)
Comment (elysee15) [reply .github/workflows/build.yml]: Okay

File: .github/workflows/build.yml (hunk 3 above)
Comment (elysee15) [reply .github/workflows/build.yml]: No I think node setup is missing

File: .github/workflows/build.yml (hunk 4 above)
Comment (elysee15) [reply .github/workflows/build.yml]: Alright

**Pull request merged**